platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/MutableRosterEntry.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.roster;
+
+import com.swirlds.common.platform.NodeId;
+import java.security.cert.X509Certificate;
+
+public interface MutableRosterEntry extends RosterEntry {
+    /**
+     * Sets the node ID of this roster entry.
+     *
+     * @param nodeId the node ID to set
+     * @return this mutable roster entry
+     */
+    MutableRosterEntry setNodeId(NodeId nodeId);
+
+    /**
+     * Sets the consensus weight of this roster entry.
+     *
+     * @param weight the consensus weight to set
+     * @return this mutable roster entry
+     */
+    MutableRosterEntry setWeight(long weight);
+
+    /**
+     * Sets the internal hostname of this roster entry.
+     *
+     * @param hostname the internal hostname to set
+     * @return this mutable roster entry
+     */
+    MutableRosterEntry setInternalHostname(String hostname);
+
+    /**
+     * Sets the internal port of this roster entry.
+     *
+     * @param port the internal port to set
+     * @return this mutable roster entry
+     */
+    MutableRosterEntry setInternalPort(int port);
```

Comment:
Before we commit to this API, let's double check to see if we even need the internal IP address and port

Reply:
Agreed, I think the internal IP address can go away.   We better decide this quick because services is baking it into their HAPI API.